crates/release_plz_core/src/git/backend.rs

```diff
@@ -579,7 +600,93 @@ impl GitClient {
+
+                // categorize the labels
+                for label in labels {
+                    match label_map.get(label) {
+                        Some(id) => label_ids.push(id.to_owned()),
+                        None => labels_to_create.push(label),
+                    }
+                }
+
+                // create missing labels
+                for label in labels_to_create {
+                    info!("Backend Gitea Creating label: {}", label);
+                    let res = self
+                        .client
+                        .post(format!("{}/labels", self.repo_url()))
+                        .json(&json!({
+                            "name": label.trim(),
+                            "color": "#FFFFFF"
+                        }))
+                        .send()
+                        .await?
+                        .successful_status()
+                        .await?;
+
+                    match res.status() {
+                        StatusCode::CREATED => {
+                            let new_label: Label = res.json().await?;
+                            label_ids.push(new_label.id);
+                        }
+                        StatusCode::NOT_FOUND => {
+                            anyhow::bail!(
+                                "Failed to create label '{}'. \n
+                            Please check if the repository URL '{}'
+                            is correct and the user has the necessary permissions..",
+                                label,
+                                self.repo_url()
+                            );
+                        }
+                        StatusCode::UNPROCESSABLE_ENTITY => anyhow::bail!(
+                            "Label '{}' creation failed and existing are {:?}",
+                            label,
+                            label_map
+                        ),
+                        _ => {
+                            anyhow::bail!(
+                                "Label creation failed response is {} \n\
+                            With Status Code: {}",
+                                label,
+                                res.status()
+                            );
+                        }
+                    }
+                }
+
+                // add all labels to PR
+                if label_ids.is_empty() {
+                    anyhow::bail!(
+                        "The provided labels: {:?} \n
+                        were not added to PR #{}",
+                        labels,
+                        pr_number
+                    );
+                } else {
+                    self.client
+                        .post(format!("{}/{}/labels", self.issues_url(), pr_number))
+                        .json(&json!({ "labels": label_ids }))
+                        .send()
+                        .await?
+                        .successful_status()
+                        .await?;
+                }
+
```

Comment:
can you extract this code in a separate function (or multiple)?

Reply:
Pushed refactor for this